test/llvm-intrinsics/llvm-vector-reduce/add.ll

```diff
@@ -0,0 +1,552 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck --check-prefix CHECK-SPIRV %s
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
+
+target triple = "spir64-unknown-unknown"
+
+; CHECK-SPIRV-DAG: TypeInt [[#I8TYPE:]] 8
```

Comment:
I do appreciate the thoroughness of this test. But I am not sure if it is really necessary to check for all data types here. If the implementation passes/fails for a particular data type, is there a reason why it will have a different result for some other data type?

Reply:
IMO there is. The major necessity here comes from the fact that translator during translation is a stateful machine. This actually requires from the test to contain at least some kind of combination of types to be sure that at least in some ways this stateful behavior is covered by tests. The second reason is to be 100% sure that the types aren't changing of the implementation behavior. And third (imo. weakest) is that the api calls which are used by implementation are also tested for correctness in that context.

Reply:
I am not sure about correctness as we are not actually running anything and creating results in the LIT tests. My argument is this. LIT tests that are being added here test specifically the sequence of SPIR-V instructions generated by 'your' implementation and if your implementation breaks one test that checks SPIR-V code generation for one type, it is going to break for all types. 

Also, I am ok to have this discussion offline and not block this PR.

Thanks